src/lantern/src/JITExecute.cpp

```diff
@@ -0,0 +1,185 @@
+  }
+  const auto& op {ops.front()};
+  return make_raw::FunctionSchema(op->schema());
+  LANTERN_FUNCTION_END_RET(nullptr)
+}
+
+// Returns all FunctionSchemas for a specific operator.
+void* _lantern_jit_all_schemas_for(void* name) {
+  LANTERN_FUNCTION_START
+  const auto& op_name {c10::Symbol::fromQualString(from_raw::string(name))};
+  const auto& ops {torch::jit::getAllOperatorsFor(op_name)};
+  std::vector<c10::FunctionSchema> schemas;
+  for (const auto& op : ops) {
+    schemas.push_back(op->schema());
+  }
+  return make_raw::vector::FunctionSchema(schemas);
+  LANTERN_FUNCTION_END_RET(nullptr)
+}
+
+// Executes the requested operation using the first FunctionSchema that matches the given arguments.
+void* _lantern_jit_execute(void* name, void* stack) {
+  LANTERN_FUNCTION_START
+  const auto& op_name {c10::Symbol::fromQualString(from_raw::string(name))};
+  const auto& ops {torch::jit::getAllOperatorsFor(op_name)};
+  const auto length {ops.size()};
+  auto& stack_ {*reinterpret_cast<torch::jit::Stack*>(stack)};
+  if (length <= 0) {
+    throw std::runtime_error("Operator not found.");
+  } else if (length == 1) {
+    const auto& op {ops.front()};
+    op->getOperation()(stack_);
+  } else {
+    LLOG("Found several matches; looking for first viable schema.\n")
+    int tries {0};
+    bool found {false};
+    for (const auto& op : ops) {
+      const auto& schema {op->schema()};
+      const auto& arguments {schema.arguments()};
+      const auto num_args {arguments.size()};
+      const auto num_given {stack_.size()};
+      if (num_given != num_args) {
+        LLOG("Schema skipped: Number of arguments given does not match expected.\n")
+        tries++;
+        continue;
+      }
+      try {
+        op->getOperation()(stack_);
+        tries++;
+        found = true;
+        break;
+      } catch (c10::Error e) {
+        const auto& schema {op->schema()};
+        const auto& arguments {schema.arguments()};
+        const auto& returns {schema.returns()};
+        std::string info{"Couldn't call operator, schema is:\n"};
+        info += "Arguments:\n";
+        for (const auto& arg : arguments) {
+          info += "  Name: " + std::string(arg.name()) + "\n";
+          info += "  Type: " + std::string(c10::typeKindToString(arg.type()->kind())) + "\n";
+        }
+        info += "Returns:\n";
+        for (const auto& ret : returns) {
+          info += "  Type: " + std::string(c10::typeKindToString(ret.type()->kind())) + "\n";
+        }
+        info += "C10 error is: " + std::string(e.what_without_backtrace());
+        info += "\nTrying next.\n";
+        LLOG(info.c_str())
+        std::cout << info; // tbd remove at some later time
+      }
+    }
+  if (found) {
+    LLOG(("Found matching schema in try: " + std::to_string(tries)).c_str())
+  } else {
+    // we return the original Stack (maybe change that logic)
+    LLOG("Tried all schemas; none matched.\n")
+    std::cout << "Tried all schemas; none matched.\n";
```

Comment:
We probably want to throw an error with that information. It would be weird if no error is raised but the function didn't work.